### PR TITLE
Feat/badge component

### PR DESCRIPTION
**Pull request by Kreezag**
#18

Create va-badge component, remove va-badge-count and va-chip.

Floating is not explicitly specified. it is set automatically if there is content in the default slot
Content adds through prop label or slot: badge. The slot has more priority to visible data in badge

Awaiting designer review

### CODE REVIEW

**Comment by asvae** [comment]
Please add link to issue in PR description.

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue [hunk 1]
```diff
@@ -0,0 +1,169 @@
+<template>
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Please get back all cases from count badge.

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        Empty with visible-empty prop: <va-badge visible-empty />
+      </div>
+      <div>
+        Not empty: <va-badge outline label="not empty" />
+      </div>
+      <div>
+        With Icon:
+        <va-badge>
+          <template v-slot:badge>
+              <span>
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.demo.vue]: span is likely not needed

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+<style lang="scss">
+@import '../../vuestic-sass/resources/resources';
+
+.va-badge {
+  display: inline-flex;
+  position: relative;
+}
+
+.va-badge__badge {
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.vue]: `.va-badge__content-wrapper` maybe?

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  <div
+    class="va-badge"
+   :class="badgeClass"
+  >
+    <div
+      class="va-badge__badge mr-2"
+      :style="badgeStyle"
+      v-if="!isEmpty"
+    >
+      <div class="va-badge__badge__content">
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.vue]: What's the reason to have so many nesting levels?

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  .va-badge--left.va-badge--bottom & {
+    transform: translateX(-100%);
+  }
+
+  .va-badge--bottom.va-badge--overlap & {
+    transform: translateX(-50%) translateY(-50%);
+  }
+}
+
+.va-badge__badge__content {
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.vue]: Please use standard wrapping approach. I.e.
```scss
.va-badge {
  &__content-wrapper {
    &__content {
      margin: auto;
      text-transform: uppercase;
      max-height: ($badge-font-size * $badge-line-height);
      min-height: ($badge-font-size * $badge-line-height);
      line-height: $badge-line-height;
      overflow: hidden;
      .va-badge--multiline & {
        max-height: 100%;
        overflow: auto;
       }
    }
  }
}
```

We don't really care on what kind of conventions to use. We care about using as few conventions as possible.

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Sorry, color text selected in color picker in demo component

[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: I fixed it. I added static cases to demo, and I move universal switches to the bottom side of the demo file

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    <VbCard title="controls">
+      TextColor prop control
+      <va-advanced-color-picker v-model="textColor"/>
+      {{ textColor }}
+    </VbCard>
+    <VbCard title="controls">
+      Transparent Slider
+      <va-slider
+        range
+        step.number="1"
```
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: this component removed from VaBadge.demo.vue

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Fixed it

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue [hunk 7]
```diff
@@ -0,0 +1,169 @@
+<template>
+  <VbDemo>
+    <VbCard title="controls">
+      Label: <va-input v-model="label"/>
```
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: fixed it

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: This case is removed.

[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Fixed it

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+import approximateNumber from 'approximate-number'
+
+export default {
+  name: 'va-badge',
+  mixins: [ColorThemeMixin],
+  props: {
+    outline: {
+      type: Boolean,
+      default: false,
+      required: false,
```
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: Fixed it

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue (hunk 2 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Fixed it

File: src/components/vuestic-components/va-badge/VaBadge.demo.vue (hunk 1 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.demo.vue]: Fixed

File: src/components/vuestic-components/va-badge/VaBadge.vue (hunk 3 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: No, it is correct. For example, vuetify have a similar realization badge component
https://vuetifyjs.com/en/components/badges#badges

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+        return false
+      }
+
+      return true
+    },
+    isFloating () {
+      return this.$slots.default || this.dot
+    },
+    formattedLabel () {
+      const isApproximateNumber = () => this.approximateNumber && this.label && !isNaN(Number(this.label))
```
[the remark this answers is not in the document]
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: fixed it

File: src/components/vuestic-components/va-badge/VaBadge.vue (hunk 4 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: Vuetify have a similar realization badge component
Example: https://vuetifyjs.com/en/components/badges#badges

Comment (asvae) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: @Kreezag I see only 2 levels of nesting in there.

File: src/components/vuestic-components/va-badge/VaBadge.vue (hunk 3 above)
Comment (asvae) [reply src/components/vuestic-components/va-badge/VaBadge.vue]: If vuetify did something wrongly doesn't mean we have to do the same.

badge.badge - makes no sense to me whatsoever.

File: src/components/vuestic-components/va-badge/VaBadge.vue [hunk 10]
```diff
@@ -0,0 +1,230 @@
+<template>
+  <div
+    class="va-badge"
+   :class="badgeClass"
+  >
+    <div
+      class="va-badge__badge mr-2"
```
Comment (asvae) [inline src/components/vuestic-components/va-badge/VaBadge.vue]: Please use different naming. Other words exist.

va-badge__external-container
va-badge__internal-container

etc.

**Pull request merged**